core/config/value.go

```diff
@@ -479,23 +480,34 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, boo
 
 					item := destSlice.Index(ai)
 					item.Set(reflect.ValueOf(itemValue))
-					found = true
 				} else {
 					break
 				}
 			}
 			if destSlice.Len() > 0 {
 				fieldValue.Set(destSlice)
 			}
-		}
-
-	}
+		case bucketMap:
+			destMap := reflect.MakeMap(fieldType).Interface()
+			val := global.GetValue(childKey).Value()
 
-	if found {
-		if errs := validator.Validate(target); errs != nil {
-			return nil, true, errs
+			if val != nil {
+				switch v := val.(type) {
+				case map[interface{}]interface{}:
+					for key, value := range v {
+						switch key := key.(type) {
+						case string:
+							destMap := destMap.(map[string]interface{})
+							destMap[key] = value
+						default:
+							return nil, fmt.Errorf("Can't populate struct from the input yaml. Map key must be a string: %v", key)
```

Comment:
nit: errors are always lowercase, to be interpolated into sentences. "can't populate struct"...